File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        
        HashMap<Character, Integer> map = new HashMap<>();
        int answer [] = new int[s.length()];
        
        for(int i=0; i<s.length(); i++){
            char a = s.charAt(i);
            int index = map.getOrDefault(a, -1);
            map.put(a, i);
            if(index == -1){
                answer[i] = index;
            }else{
                answer[i] = i-index;
            }         
        }

        return answer;
    }
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    
    const map = new Map();
    var answer = [];
    
    for(let i=0; i<s.length; i++){
        var char = s.charAt(i);
        if(!map.has(char)){
            answer.push(-1);
        }else{
            var index = map.get(char);
            answer.push(i-index);
        }
        map.set(char, i);
    }

    return answer;
}